lib/rule-data-snapshot.js

```diff
@@ -3,6 +3,14 @@
 
 var merge = require('lodash.mergewith');
 
+function isEmptyObj (obj) {
+  return obj && (typeof obj == "object") && Object.keys(obj).length === 0 && obj.constructor === Object
+}
+
+function coerce(v) {
+  return (isEmptyObj(v) ? null : v)
```

Comment:
I would prefer a semicolon.and no parenthesis; more consistent with this script style.